src/AspNet.Security.OAuth.QQ/QQAuthenticationHandler.cs

```diff
@@ -0,0 +1,127 @@
+                    .AddOptionalClaim(ClaimTypes.Name, QQAuthenticationHelper.GetNickname(payload), Options.ClaimsIssuer)
+                    .AddOptionalClaim(ClaimTypes.Gender, QQAuthenticationHelper.GetGender(payload), Options.ClaimsIssuer)
+                    .AddOptionalClaim("urn:qq:figureurl", QQAuthenticationHelper.GetFigureUrl(payload), Options.ClaimsIssuer)
+                    .AddOptionalClaim("urn:qq:figureurl_1", QQAuthenticationHelper.GetFigureUrl1(payload), Options.ClaimsIssuer)
+                    .AddOptionalClaim("urn:qq:figureurl_2", QQAuthenticationHelper.GetFigureUrl2(payload), Options.ClaimsIssuer)
+                    .AddOptionalClaim("urn:qq:figureurl_qq_1", QQAuthenticationHelper.GetFigureQQUrl1(payload), Options.ClaimsIssuer)
+                    .AddOptionalClaim("urn:qq:figureurl_qq_2", QQAuthenticationHelper.GetFigureQQUrl2(payload), Options.ClaimsIssuer);
+
+            var principal = new ClaimsPrincipal(identity);
+            var ticket = new AuthenticationTicket(principal, properties, Options.AuthenticationScheme);
+
+            var context = new OAuthCreatingTicketContext(ticket, Context, Options, Backchannel, tokens, payload);
+            await Options.Events.CreatingTicket(context);
+
+            return context.Ticket;
+        }
+
+        protected override async Task<OAuthTokenResponse> ExchangeCodeAsync(string code, string redirectUri)
+        {
+            var queryString = new Dictionary<string, string>()
+            {
+                {"client_id",Options.ClientId},
+                {"client_secret",Options.ClientSecret },
+                {"redirect_uri",redirectUri },
+                {"code",code },
+                {"grant_type","authorization_code" },
+            };
+            var endpoint = QueryHelpers.AddQueryString(Options.TokenEndpoint, queryString);
+            var response = await Backchannel.GetAsync(endpoint);
+            if (!response.IsSuccessStatusCode)
+            {
+                var error = "OAuth token endpoint failure: " + await response.Content.ReadAsStringAsync();
+                return OAuthTokenResponse.Failed(new Exception(error));
+            }
+            var payload = JObject.FromObject(QueryHelpers.ParseQuery(await response.Content.ReadAsStringAsync()).ToDictionary(k => k.Key, k => k.Value.ToString()));
+            return OAuthTokenResponse.Success(payload);
+        }
+
+        private async Task<string> GetUserOpenIdAsync(OAuthTokenResponse tokens)
+        {
+            var endpoint = QueryHelpers.AddQueryString(Options.OpenIdEndpoint, "access_token", tokens.AccessToken);
+            var response = await Backchannel.GetAsync(endpoint);
+            if (!response.IsSuccessStatusCode)
+            {
+                throw new HttpRequestException($"An error occurred while retrieving the user openid.({response.StatusCode})");
+            }
+            var body = await response.Content.ReadAsStringAsync();
+            var i = body.IndexOf("{");
```

Comment:
The QQ OAuth server responses openid content format like this: `callback( {“client_id”:”YOUR_APPID”,”openid”:”YOUR_OPENID”} );`